### ISSUE TITLE
Sometimes emit lane information in instructions

### ISSUE THREAD

**Issue by freenerd** [Project-OSRM/osrm-text-instructions#48]
Currently we only emit [lane information](https://github.com/Project-OSRM/osrm-text-instructions/blob/8ea67611387c52771a6fd40918030b67ccf55179/instructions.json#L36-L40) during `type: use lane`.

We should do that in other cases as well. Most cases we don't need to do that (e.g. on simple forks or turns) but identify cases like ["stay right to turn left"](https://github.com/Project-OSRM/osrm-text-instructions/issues/42) where it will make sense.

**Comment by 1ec5** [comment]
Here are some ideas for deciding when to issue lane guidance, in no particular order:

* Issue lane guidance if multiple lanes are valid.
* Do not issue lane guidance if all the lanes are valid.
* Issue lane guidance if the valid lanes are counterintuitive (as in a [hook turn](https://en.wikipedia.org/wiki/Hook_turn)).
* Issue lane guidance if turning left but the leftmost lane is invalid (such as if the leftmost lane is reserved for U-turns).
* Issue lane guidance if the intersection distinguishes between a slight right turn and a right turn.
* Do not issue lane guidance if there are fewer than three lanes.

**Comment by yuryleb** [comment]
I'd like to exclude _"Do not issue lane guidance if there are fewer than three lanes"_ bullet, sometimes lane gudance could be non-trivial even for 2 lanes road :wink: 
I'm sorry for photo quality in motion on [this street](https://www.openstreetmap.org/way/568501507):
![vdo_7776_0309_ 00_00_00](https://user-images.githubusercontent.com/4529411/37242892-88c34a02-2468-11e8-8d25-318757dd971a.jpg)

**Comment by 1ec5** [comment]
True, I guess in that case lane guidance would be extraneous for turning right but worth making explicit for turning left or going straight.
